File: ios/GrowthbeatStarterProject/GrowthbeatStarterProject/ViewController.m
//
//  ViewController.m
//  GrowthbeatStarterProject
//
//  Created by SIROK, Inc. on 04/03/2016.
//  Copyright © 2016 SIROK, Inc. All rights reserved.
//

#import "ViewController.h"
#import <Growthbeat/Growthbeat.h>

@interface ViewController ()

@end

@implementation ViewController

@synthesize developmentTagSwitch;
@synthesize levelTextField;
@synthesize itemTextField;
@synthesize priceTextField;

- (void)viewDidLoad {
    [super viewDidLoad];
    [self initView];
}

- (IBAction)randomBtnClicked:(id)sender
{
     [[GrowthAnalytics sharedInstance] setRandom];
}

- (IBAction)tapLevelTagButton:(id)sender
{
    [[GrowthAnalytics sharedInstance] setLevel:[levelTextField.text intValue]];
}

- (IBAction)switchEnvironment:(id)sender
{
    [[GrowthAnalytics sharedInstance] setDevelopment:developmentTagSwitch.on];
}

- (IBAction) tapPurchaseEventButton:(id)sender {
    [[GrowthAnalytics sharedInstance] purchase:[priceTextField.text intValue] setCategory:@"item" setProduct:itemTextField.text];
}

- (BOOL)textFieldShouldReturn:(UITextField *)textField
{
    [textField resignFirstResponder];
    return YES;
}

-(void) initView
{
    [self.view.layer setCornerRadius:7.0];
    [self.view.layer setMasksToBounds:YES];
    self.view.layer.opaque = NO;
}

@end

File: ios/GrowthbeatStarterProject-Swift/GrowthbeatStarterProject/ViewController.swift
//
//  ViewController.swift
//  GrowthbeatStarterProject
//
//  Created by Sota Yamashita on 22/03/2016.
//  Copyright © 2016 Sota Yamashita. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        self.initView()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
    func initView() {
        self.view.layer.cornerRadius = 7.0
        self.view.layer.opaque = false
    }
    
    @IBAction func randomBtnClicked(sender:AnyObject) {
        GrowthAnalytics.sharedInstance().setRandom();
    }

    @IBOutlet weak var levelTextField: UITextField?
    @IBAction func tapLevelTagButton(sender:AnyObject) {
        GrowthAnalytics.sharedInstance().setLevel((levelTextField!.text! as NSString).intValue)
    }
    
    @IBOutlet weak var developmentTagSwitch: UISwitch?
    @IBAction func switchEnvironment(sender:AnyObject) {
        GrowthAnalytics.sharedInstance().setDevelopment((developmentTagSwitch?.on)!);
    }
    
    @IBOutlet weak var priceTextField: UITextField?
    @IBOutlet weak var itemTextField: UITextField?
    @IBAction func tapPurchaseEventButton(sender:AnyObject) {
        GrowthAnalytics.sharedInstance().purchase(
            (priceTextField!.text! as NSString).intValue,
            setCategory: "item",
            setProduct: itemTextField?.text
        )
    }
}

